### DISCUSSION TITLE
Add option to always use TTF fonts (and configure which fonts to use), even with Latin-based languages

### DISCUSSION THREAD

**Discussion by RQWorldblender** [OpenRCT2/OpenRCT2#15747] [Feature requests]
When playing on a large screen resolution, at least full HD resolution (1920x1080), and if using the original UI scaling size, the sprite fonts are often hard to see. I could increase the UI scaling, but it will cause most of the UI elements to become pixelated if increased to a large enough size. If I could use TTF fonts here, I would feel more confident about increasing the UI scaling.

Rollercoaster Tycoon Classic does not use the original sprite fonts; it likely uses TTF fonts for almost all of its UI. I know TTF fonts can be used in some situations (for languages not originally supported by RCT2), but I would like to use TTF fonts even if there sprite fonts available. TTF fonts are more likely to scale smoothly, without becoming pixellated, allowing me and other people to comfortably see almost all of the in-game text.

This issue is not about the other sprites that are drawn on the map, since those are harder to replace.

**Comment by Gymnasiast** [comment]
> TTF fonts are more likely to scale smoothly, without becoming pixellated

You haven't seen our TTF rendering, then. :-(

At the moment, this feature would be useless. TTF text is not only ugly, it also does not work in OpenGL mode.

**Comment by RQWorldblender** [comment]
> You haven't seen our TTF rendering, then. :-(
> 
> At the moment, this feature would be useless. TTF text is not only ugly, it also does not work in OpenGL mode.

Oh well. It makes me think I should work on this feature then.

**Comment by Ruedii** [comment]
Also, using a single UTF-8 font and internally using UTF-8 on the new save format might be useful.  However, this MIGHT take some time.

Part of the problem is we need good TTF fonts, that still look "retro".   Balancing the old style pixel font drawing with vertex rendering and anti-aliasing can be a tricky work of art.

If we want a non-retro font available, that would work just fine.  It's just the retro one could prove tricky.  Might be a good thing to have as a menu option.

**Comment by RQWorldblender** [comment]
> Part of the problem is we need good TTF fonts, that still look "retro". Balancing the old style pixel font drawing with vertex rendering and anti-aliasing can be a tricky work of art.
> 
> If we want a non-retro font available, that would work just fine. It's just the retro one could prove tricky. Might be a good thing to have as a menu option.

I'm okay with using non-retro fonts. The thing I care more about is being able to read the text without me having to squint so often. I mentioned in an earlier comment that Rollercoaster Tycoon Classic doesn't use the original pixel font at all. It's only using non-retro fonts for almost all of its UI.
